internal/services/openai.go

```diff
@@ -0,0 +1,117 @@
+package services
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/DIMO-Network/devices-api/internal/config"
+	"github.com/pkg/errors"
+)
+
+type OpenAI interface {
+	QueryDeviceErrorCodes(make, model string, year int32, errorCodes []string) (string, error)
+}
+
+type openAi struct {
+	baseURL    string
+	token      string
+	httpClient *http.Client
+}
+
+type ChatGptResponseChoices struct {
+	Message struct {
+		Role    string
+		Content string
+	}
+	Index        int
+	FinishReason string `json:"finish_reason"`
+}
+
+type ChatGptResponse struct {
```

Comment:
```suggestion
type ChatGPTResponse struct {
```
Does this need to be public?